lib/errors/error-collector.js

```diff
@@ -89,69 +89,47 @@ class ErrorCollector {
   }
 
   /**
-   * Helper method for processing errors that are created with .noticeError()
+   * Gets the iterable property from the transaction based on the error type
+   * @param {Transaction} transaction the collected exception's transaction
+   * @param {string} errorType the type of error to be processed: "user", "transactionException", "transaction"
+   * @returns
+   */
+  _getIterableProperty(transaction, errorType) {
+    let iterableProperty = null
+    if (errorType === 'user' && transaction.userErrors.length) {
+      iterableProperty = transaction.userErrors
+    }
+    if (errorType === 'transactionException') {
+      iterableProperty = transaction.exceptions
+    }
+    return iterableProperty
+  }
+
+  /**
+   * Helper method for processing errors that are created with .noticeError(), exceptions
+   * on the transaction (transaction.exceptions array), and inferred errors based on Transaction metadata.
    *
    * @param {Transaction} transaction the collected exception's transaction
    * @param {number} collectedErrors the number of errors we've successfully .collect()-ed
    * @param {number} expectedErrors the number of errors marked as expected in noticeError
-   * @returns {Array.<number>} the updated [collectedErrors, expectedErrors] numbers post-processing
+   * @param {string} errorType the type of error to be processed; "user", "transactionException", "transaction"
+   * @returns {Array.<number>} the updated [collectedErrors, expectedErrors] numbers post processing
    */
-  _processUserErrors(transaction, collectedErrors, expectedErrors) {
-    if (transaction.userErrors.length) {
-      for (let i = 0; i < transaction.userErrors.length; i++) {
-        const exception = transaction.userErrors[i]
+  _processErrors(transaction, collectedErrors, expectedErrors, errorType) {
+    const iterableProperty = this._getIterableProperty(transaction, errorType)
+    if (iterableProperty) {
+      iterableProperty.forEach((exception) => {
```

Comment:
I'm not sure if we are in a "hot path" here, so it may not be worth a refactor, but I'd like to call attention to the performance difference between the options for looping here.

Utilizing this benchmark script:

```js
'use strict'

const { Bench } = require('tinybench')
const bench = new Bench({ time: 100 })

const input = new Array(1_000_000)
bench
	.add('traditional for', function () {
		let x
		for (let i = 0; i < input.length; i += 1) {
			x = input[i]
		}
	})
	.add('for...of', function () {
		let x
		for (const i of input) {
			x = i
		}
	})
	.add('forEach', function () {
		let x
		input.forEach((i) => {
			x = i
		})
	})

bench
	.warmup()
	.then(() => bench.run())
	.then(() => {
		console.table(bench.table())
	})
```

We can see a result set like:
```
┌─────────┬───────────────────┬─────────┬────────────────────┬──────────┬─────────┐
│ (index) │ Task Name         │ ops/sec │ Average Time (ns)  │ Margin   │ Samples │
├─────────┼───────────────────┼─────────┼────────────────────┼──────────┼─────────┤
│ 0       │ 'traditional for' │ '2,042' │ 489546.9463414662  │ '±0.74%' │ 205     │
│ 1       │ 'for...of'        │ '1,576' │ 634174.5822784823  │ '±0.27%' │ 158     │
│ 2       │ 'forEach'         │ '773'   │ 1293309.2948717843 │ '±0.64%' │ 78      │
└─────────┴───────────────────┴─────────┴────────────────────┴──────────┴─────────┘
```

I ran this test with Node.js 20.15.0 on my M1 Max based system. The `forEach` is the slowest of the methods due to the additional cost of invoking a function on each iteration. Consider how `forEach` might be implemented:

```js
Array.prototype.forEach = function (iterator) {
	for (let i = 0; i < this.length; i += 1) {
		iterator(this[i])
	}
}
```

In conclusion, `forEach` is a nice convenience tool but _may_ not always be appropriate.